lib/datasource/npm/index.ts

```diff
@@ -5,3 +5,4 @@ export { getReleases } from './releases';
 export { getNpmrc, setNpmrc } from './npmrc';
 export { id } from './common';
 export const defaultVersioning = npmVersioning.id;
+export const registryUrlRestriction = 'disallowed';
```

Comment:
This needs refactoring later to use `defaultRegistryUrls` and `rangeStrategy=first`. But first we need to stop using `.npmrc` directly

Reply:
Doesn't this now blocks custom npm registries? I'm overriding this on many internal projects.

Reply:
Wasn't intentional and can be reverted.

Reply:
I checked and there's no `registryUrls` use directly in lib/datasource/npm. What approach do you use to override it?

Reply:
Ok, so how get the registry url passed from npmrc of repo?

Reply:
There's an npm lib that actually reads from disk as part of the npm datasource

Reply:
So maybe we should move that logic to manager in a future pr.